app/uk/gov/hmrc/hec/repos/HECTaxCheckStore.scala

```diff
@@ -218,6 +219,46 @@ class HECTaxCheckStoreImpl @Inject() (
               logger.warn(s"${invalid.size} results failed json parsing - $errorStr")
             }
 
+            Either.cond(invalid.isEmpty, valid, Error(Left(errorStr)))
+          }
+      }
+    )
+
+  private def findByLimit[T](
+    fieldName: String,
+    value: T,
+    skipN: Int,
+    limitN: Int,
+    sortByFieldName: String
+  ): EitherT[Future, Error, List[HECTaxCheck]] =
+    EitherT(
+      preservingMdc {
+        collection
+          .find(Filters.equal[T](fieldName, value))
+          .sort(Sorts.ascending(sortByFieldName))
+          .skip(skipN)
+          .limit(limitN)
+          .toFuture()
+          .map { caches =>
```

Comment:
the thing inside this map is the same as the thing above - we could factor it out in a `def` like
```scala
def f(caches: Seq[CacheItem]): Either[Error, List[HECTaxCheck]]
```
perhaps to avoid too much duplication

Reply:
[updated](https://github.com/hmrc/hec/pull/48/commits/6ea63bbe06cfb1cb90a7034a1cce22b8ce32882b)